### PR TITLE
feat: rate-limit konnect requests when receiving 429s

### PR DESCRIPTION
**Pull request by GGabriele**

### CODE REVIEW

File: go.mod
Comment (hbagdi) [inline go.mod:38]: Please move this into the first require block that starts at line 9.

File: utils/types.go
Comment (hbagdi) [inline utils/types.go:183]:
```suggestion
		minRetryWait = 10 * time.Second
		maxRetryWait = 60 * time.Second
		retryMax     = 10
```

Let's start with a bit more conservative values that we can tune later on.

**Review by hbagdi — COMMENTED** [review]
Please get a +1 from @tjasko.

File: utils/types.go
Comment (tjasko) [inline utils/types.go:136]: Let's update this link so that it doesn't go out of date:
https://github.com/hashicorp/go-retryablehttp/blob/v0.7.1/client.go#L503

File: utils/types.go
Comment (tjasko) [inline utils/types.go:145]: nit: Can be optimized to:
```suggestion
	if resp != nil && resp.StatusCode == http.StatusTooManyRequests {
```

**Review by tjasko — COMMENTED** [review]
Changes seem sane, however, aren't we waiting on hashicorp/go-retryablehttp#168 to get merged?

**Comment by GGabriele** [comment]
> Changes seem sane, however, aren't we waiting on [hashicorp/go-retryablehttp#168](https://github.com/hashicorp/go-retryablehttp/pull/168) to get merged?

That's not a blocker, because the way decK uses the underlying library via `go-kong` cannot hit that corner case.

**Review by tjasko — DISMISSED** [review]

**Pull request merged**
